### DISCUSSION TITLE
Smoke Test Flag

### DISCUSSION THREAD

**Discussion by kateefimova-eb** [storybookjs/storybook#15952] [General] [answered]
Hi folks! :wave: Anyone here closely familiar with --smoke-test flag? I seem to be getting an exit code of 1, regardless if storybook is in the good state(with a couple of warnings) or in a broken state(where it fails during the start process). Is this by design, or should successful state return w code 0 and there's something up with the code?

**Comment by shilman** [comment, marked as answer]
here's literally what it does:

```
    process.exit(
      managerWarnings.length > 0 || (previewWarnings.length > 0 && !options.ignorePreview) ? 1 : 0
    );
```

where managerWarnings/previewWarnings are webpack warnings. So you could run yarn storybook and take a look at the warnings which are causing the smoke test to fail.

i'm also working on a test runner that's part of our interaction testing project. that will go through and try to render each story and fail if any of the stories fail, which is a slightly more thorough test. ETA 6.5 release
